[Merged by Bors] - refactor(data/finset/lattice): sup' and inf' without option

pechersky opened the pull request:
Hide the option API further, by using `unbot` and `untop`.



---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message 
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

pechersky committed:
refactor(data/finset/lattice): sup' and inf' without option

Hide the option API further, by using `unbot` and `untop`.

pechersky committed:
More mentions of hypotheses

eric-wieser commented on src/data/finset/basic.lean:
```diff
@@ -1990,7 +1990,7 @@ eq_of_veq $ multiset.map_cons f a s.val
 by rw [nonempty_iff_ne_empty, nonempty_iff_ne_empty, ne.def, map_eq_empty]
 
-alias map_nonempty ↔ _ nonempty.map
+alias map_nonempty ↔ nonempty.comap nonempty.map
```
I'm not really a fan of this name, it suggests a `finset.comap`.

eric-wieser commented on src/data/finset/lattice.lean:
```diff
@@ -580,6 +580,11 @@ begin
 end
 
+@[simp] lemma sup'_map {s : finset γ} {f : γ ↪ β} (g : β → α) (hs : (s.map f).nonempty)
+  (hs': s.nonempty := hs.comap) :
```
```suggestion
  (hs' : s.nonempty := finset.map_nonempty.mp h) :
```
(Missing space, and avoid introducing the questionable alias)

eric-wieser reviewed (approved):
bors d+

Thanks!

pechersky committed:
remove alias renaming because of suggestion

pechersky committed:
Merge branch 'pechersky/finset-sup-prime-no-option' of https://github.com/leanprover-community/mathlib into pechersky/finset-sup-prime-no-option

pechersky committed:
Bring back broken proof, oops

pechersky committed:
Merge branch 'master' into pechersky/finset-sup-prime-no-option

pechersky committed:
Redefine max' and min' to be based on sup' and inf'

eric-wieser commented on src/data/finset/lattice.lean:
```diff
@@ -544,15 +544,15 @@ lemma comp_sup'_eq_sup'_comp [semilattice_sup γ] {s : finset β} (H : s.nonempt
-  { rw [with_bot.none_eq_bot, bot_sup_eq], exact bot_sup_eq.symm, },
-  { cases f₂, refl,
-    exact congr_arg coe (g_sup f₁ f₂), },
+  refine f₁.rec_bot_coe _ _,
```
Does 
```suggestion
  induction f₁ using with_bot.rec_bot_coe with f₁,
```
work here? (I migght have the `using` and `with` reversed.

eric-wieser commented on src/data/finset/lattice.lean:
```diff
@@ -877,19 +887,24 @@ le_inf st
 
 variables (s : finset α) (H : s.nonempty) {x : α}
 
-theorem min'_mem : s.min' H ∈ s := mem_of_min $ by simp [min']
```
How was `simp` proving this before?

pechersky replied:
Brought `simp` back.

pechersky committed:
Simplify `min'_mem` proof

Goes through `coe_inf'`, have to unwrap defs to get back to a place where
`coe o id` simplifies to `coe`.

eric-wieser commented on src/data/finset/lattice.lean:
```diff
@@ -544,15 +544,16 @@ lemma comp_sup'_eq_sup'_comp [semilattice_sup γ] {s : finset β} (H : s.nonempt
+  { rw [bot_sup_eq], exact bot_sup_eq.symm, },
+  { induction f₂ using with_bot.rec_bot_coe,
+    { simp [g'] },
+    { simp [g', ←with_bot.coe_sup, g_sup] } }
```
I'm a bit surprised the `exact` and `refl` had to be replaced with `simp` here.

eric-wieser reviewed (approved):
LGTM, it's already delegated. Was the change from `refl` to `simp` necessary  and/or deliberate?

pechersky committed:
bring old proofs back
